packages/renderer/src/assets/download-and-map-assets-to-file.ts

```diff
@@ -63,8 +62,18 @@ export const downloadAndMapAssetsToFileUrl = async ({
 	webpackBundle: string;
 	onDownload: (src: string) => void;
 }): Promise<TAsset> => {
-	const {pathname} = new URL(localhostAsset.src);
-	const newSrc = path.join(webpackBundle, sanitizeFilename(pathname));
+	const {pathname, search} = new URL(localhostAsset.src);
+
+	const split = pathname.split('.');
+	const fileExtension =
+		split.length > 1 && split[split.length - 1]
+			? `.${split[split.length - 1]}`
+			: '';
```

Comment:
when using `/speech?text=blahblahblah` like tts API , this will have no file extension.

Reply:
can we possibly identify the remote audio assets file extension from the response `Content-type` header ?